app/controllers/responsible_body/donated_devices/interest_controller.rb

```diff
@@ -0,0 +1,227 @@
+class ResponsibleBody::DonatedDevices::InterestController < ResponsibleBody::BaseController
+  before_action :redirect_if_already_completed, except: %i[select_schools opted_in]
+  before_action :find_request!, only: %i[select_schools device_types how_many_devices address disclaimer check_answers]
+
+  def new
+    @form = DonatedDeviceInterestForm.new
+  end
+
+  def create
+    @form = DonatedDeviceInterestForm.new(device_interest_params)
+
+    if @form.valid?
+      if @form.interested?
+        redirect_to responsible_body_donated_devices_about_devices_path(@school)
+      else
+        render :not_interested
+      end
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def about; end
+
+  def queue; end
+
+  def interest_confirmation
+    @form = DonatedDeviceInterestForm.new(device_interest_params)
+    if request.post?
+      if @form.valid?
+        if @form.interested?
+          redirect_to responsible_body_donated_devices_all_or_some_schools_path
+        else
+          render :not_interested
+        end
+      else
+        render :interest_confirmation, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def all_or_some_schools
```

Comment:
This method is pretty long, and a bit tricky to follow all the ifs/elses - could some of the cases be factored out to a separate method?